### PR TITLE
Jose wills code review

### PR DESCRIPTION
**Pull request by StuartCowley**

### CODE REVIEW

File: src/kata1.fizzBuzz.js [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+  }
+  if (multipleOf3) {
+    return "Fizz";
+  }
+  if (multipleOf5) {
+    return "Buzz";
+  }
+  if (!(multipleOf3 || multipleOf5)) {
+    return number;
+  }
```
Comment (StuartCowley) [inline src/kata1.fizzBuzz.js:16]: Eslint is complaining a bit here, as you do not have a return statement at the root of the function. This is what we call a default exit condition, meaning a condition that will always be hit if no other statements before this have been reached, and its generally a good practice as it means there will be no situation where no value is returned from your function

In this case all we need to do is remove the final `if`  statement and just have it as a return by itself:
```
  return number;
```

File: src/kata4.humanCatDogYears.js [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+  if (number === 1) {
+    return [number, 15, 15];
+  }
+  if (number === 2) {
+    return [number, 24, 24];
+  }
+  if (number >= 3) {
+    return [number, 24 + (number - 2) * 4, 24 + (number - 2) * 5];
+  }
+};
```
Comment (StuartCowley) [inline src/kata4.humanCatDogYears.js:11]: We have the same eslint error caused by not having a default exit condition here. As an example, what would happen if 0 was passed in? You have a working condition for any number greater than or equal to 2 here, so I would add a condition for zero at the start in addition to your others, then make the greater than 2 function your default:
```
const humanCatDogYears = (number) => {
  if (number === 0) {
    return [0, 0, 0];
  }
  if (number === 1) {
    return [number, 15, 15];
  }
  if (number === 2) {
    return [number, 24, 24];
  }
  return [number, 24 + (number - 2) * 4, 24 + (number - 2) * 5];
};
```

File: src/kata6.joinNames.js [hunk 3]
```diff
@@ -1,3 +1,13 @@
...
+  const namesArray = namesObj.map((person) => person.name);
+
+  const stringOfNames = namesArray.reduce((acc, curr, index) => {
+    if (index === namesArray.length - 1) {
+      return acc + " & " + curr;
+    }
+    return acc + ", " + curr;
+  });
+  return stringOfNames;
+};
```
Comment (StuartCowley) [inline src/kata6.joinNames.js:11]: Nice approach! Theres a couple of things to improve here, you dont need to create the variable on line 4 only to return it on line 10, you can just return the value of the function directly. Also, eslint is not happy about the formatting of lines 6 and 8, as template literal notation is preferred here. PUttin ghtese together you would end up with this:
```
const joinNames = (namesObj) => {
  const namesArray = namesObj.map((person) => person.name);

  return namesArray.reduce((acc, curr, index) => {
    if (index === namesArray.length - 1) {
      return `${acc} & ${curr}`;
    }
    return `${acc}, ${curr}`;
  });
};
```

File: src/kata7.getEmployerRole.js [hunk 4]
```diff
@@ -1,3 +1,8 @@
-const getEmployerRole = (employeeName, employees) => {};
+const getEmployerRole = (employeeName, employees) => {
+  const findEmployee = employees.find(
+    (employee) => employee.name === employeeName
+  );
+  return findEmployee.role;
```
Comment (StuartCowley) [inline src/kata7.getEmployerRole.js:5]: You can reduce this down a little bit:
```
  employees.find((employee) => employee.name === employeeName).role;
```

File: test/kata2.booleanToWord.test.js [hunk 5]
```diff
@@ -1,5 +1,10 @@
 const { booleanToWord } = require("../src");
 
 describe("booleanToWord", () => {
-  // how do we create specs again???
+  it("returns Yes when passed true", () => {
+    expect(booleanToWord(true)).toEqual("Yes");
+  });
+  it("returns No when passed false", () => {
```
Comment (StuartCowley) [inline test/kata2.booleanToWord.test.js:7]: Its a common convention to leave a clear line between each of your `it` blocks to make the test file easier to read:
```
  });

  it("returns No when passed false", () => {
```

File: test/kata4.humanCatDogYears.test.js [hunk 6]
```diff
@@ -1,3 +1,10 @@
 const { humanCatDogYears } = require("../src");
 
-// Look Ma, no handlebars!!!
+describe("humanCatDogYears", () => {
+  it("returns array of human, cat and dog years", () => {
+    expect(humanCatDogYears(1)).toEqual([1, 15, 15]);
```
Comment (StuartCowley) [inline test/kata4.humanCatDogYears.test.js:5]: Would be good to get a couple more edge cases in here, such as passing in zero of a negative number. How should the function respond in cases like this?

File: test/kata5.reachDestination.test.js [hunk 7]
```diff
@@ -1,5 +1,7 @@
 const { reachDestination } = require("../src");
 
 describe("reachDestination", () => {
-  test("returns string with estimated time of arrival", () => {});
+  it("returns string with estimated time of arrival", () => {
+    expect(reachDestination(44, 10)).toEqual("I should be there in 4.5 hours.");
```
Comment (StuartCowley) [inline test/kata5.reachDestination.test.js:5]: It would be good to add a few more examples here. How do you if the function is working properly with only one test scenario?

File: test/kata7.getEmployerRole.test.js [hunk 8]
```diff
@@ -1,5 +1,26 @@
...
+      {
+        name: "Javid",
+
+        role: "Human Recommended Reading Assistant",
+      },
+    ];
+    expect(getEmployerRole("Javid", employees)).toEqual(
+      "Human Recommended Reading Assistant"
+    );
+  });
```
Comment (StuartCowley) [inline test/kata7.getEmployerRole.test.js:25]: Small formatting change here, you dont need the clear lines inside the objects, and its good to leave a clear line between the setup part of the tests and the start of the assertions:
```
  it("returns the employee's role in the company", () => {
    const employees = [
      {
        name: "Satti",
        role: "Developer",
      },
      {
        name: "Jenny",
        role: "Sales Associate",
      },
      {
        name: "Javid",
        role: "Human Recommended Reading Assistant",
      },
    ];

    expect(getEmployerRole("Javid", employees)).toEqual(
      "Human Recommended Reading Assistant"
    );
  });
```

**Review by StuartCowley — COMMENTED** [review]
Nice work! Most of the comments are about formatting, theres a few improvements to the tests suggested as well as a couple of code improvements but nothing serious. It all works and the tests pass, so Id be pretty happy with this. Keep it up :)
